File: mawhrin/tasks/list.go
package tasks

import (
	"fmt"
	"io"

	"github.com/stvmln86/mawhrin/mawhrin/items/book"
)

// ListTask is a Task that lists Notes.
var ListTask = New(
	"list TEXT:",
	"List all notes, or notes matching TEXT.",

	func(w io.Writer, book *book.Book, amap map[string]string) error {
		notes, err := book.Match(amap["TEXT"])
		if err != nil {
			return err
		}

		for _, note := range notes {
			fmt.Fprintf(w, "%s\n", note.Name())
		}

		return nil
	},
)

File: mawhrin/comms/list.py
"""
Command definition for 'list'.
"""

import click

from mawhrin.comms.base import group
from mawhrin.items import Book


@group.command(name="list", short_help="List all notes.")
@click.argument("NAME", default="")
@click.pass_obj
def list_(book: Book, name: str):
    """
    List all notes, or notes matching NAME.
    """

    for note in book:
        if note.match(name):
            click.echo(note.name)
